core/src/core/usb.zig

```diff
@@ -610,7 +604,7 @@ pub const ConfigurationDescriptor = extern struct {
 
     pub fn serialize(self: *const @This()) [9]u8 {
         var out: [9]u8 = undefined;
-        out[0] = 9; // length
+        out[0] = out.len;
         out[1] = @intFromEnum(self.descriptor_type);
         out[2] = @intCast(self.total_length & 0xff);
         out[3] = @intCast((self.total_length >> 8) & 0xff);
```

Comment:
There's `std.mem.writeInt` which can be used to write big/little endian integers of bigger sizes into arrays